back/internal/dbg/dbg.go

```diff
@@ -0,0 +1,60 @@
+// Package dbg allows for outputting information that can help with debugging
+// the application.
+package dbg
+
+import (
+	"io"
+	"log"
+	"sync/atomic"
+)
+
+// dbg is thread-safe.
+type dbg struct {
+	log     *log.Logger
+	atomVal atomic.Value
```

Comment:
Please explore using uint32 instead as per our discussion.

Reply:
I created a PR #161 for using sync/atomic uint32 methods.